source/Body.cpp

```diff
@@ -198,84 +223,340 @@ const Government *Body::GetGovernment() const
+			spriteAnimationParameters.delay = child.Value(1);
+		else if(child.Token(0) == "transition delay" && child.Size() >= 2 && child.Value(1) > 0.)
+			spriteAnimationParameters.transitionDelay = child.Value(1);
+		else if(child.Token(0) == "scale" && child.Size() >= 2 && child.Value(1) > 0.)
+			spriteAnimationParameters.scale = static_cast<float>(child.Value(1));
+		else if(child.Token(0) == "start frame" && child.Size() >= 2)
+		{
+			spriteAnimationParameters.startFrame = static_cast<float>(child.Value(1));
+			spriteAnimationParameters.startAtZero = true;
+		}
+		else if(child.Token(0) == "indicate percentage" && child.Size() >= 2 && child.Value(1) > 0. && child.Value(1) < 1.)
+		{
+			spriteAnimationParameters.indicateReady = true;
+			spriteAnimationParameters.repeat = false;
+			spriteAnimationParameters.startAtZero = true;
+			spriteAnimationParameters.indicatePercentage = static_cast<float>(child.Value(1));
+		}
+		else if(child.Token(0) == "no indicate")
+		{
+			spriteAnimationParameters.indicateReady = false;
+		}
+		else if(child.Token(0) == "indicate")
+		{
+			spriteAnimationParameters.indicateReady = true;
+			spriteAnimationParameters.repeat = false;
+			spriteAnimationParameters.startAtZero = true;
+			spriteAnimationParameters.indicatePercentage = 1.0f;
+		}
+		else if(child.Token(0) == "random")
+			spriteAnimationParameters.randomize = true;
+		else if(child.Token(0) == "random start frame")
+			spriteAnimationParameters.randomizeStart = true;
+		else if(child.Token(0) == "no repeat")
+		{
+			spriteAnimationParameters.repeat = false;
+			spriteAnimationParameters.startAtZero = true;
+		}
+		else if(child.Token(0) == "transition rewind")
+		{
+			spriteAnimationParameters.transitionFinish = false;
+			spriteAnimationParameters.transitionRewind = true;
+		}
+		else if(child.Token(0) == "transition finish")
+		{
+			spriteAnimationParameters.transitionFinish = true;
+			spriteAnimationParameters.transitionRewind = false;
+		}
+		else if(child.Token(0) == "rewind")
+			spriteAnimationParameters.rewind = true;
+		else
+			child.PrintTrace("Skipping unrecognized attribute:");
+	}
+
+	spriteData->SetSprite(trigger, sprite, spriteAnimationParameters);
+
+	// Now load the trigger sprites.
+	for(auto node = triggerSpriteDefer.begin(); node != triggerSpriteDefer.end(); ++node){
+		// TODO: this->LoadTriggerSprite(child, state, &spriteAnimationParameters);
+	}
+
+	if(scale != 1.f)
+		GameData::GetMaskManager().RegisterScale(sprite, Scale());
+}
 
 
 // Save the sprite specification, including all animation attributes.
-void Body::SaveSprite(DataWriter &out, const string &tag) const
+void Body::SaveSprite(DataWriter &out, const string &tag, bool allStates) const
 {
-	if(!sprite)
-		return;
+	// Write all states to file
+	if(allStates){
+		std::string tags[BodyState::NUM_STATES] = {"sprite-flying", "sprite-firing", "sprite-launching", "sprite-landing", "sprite-jumping", "sprite-disabled"};
```

Comment:
Wrap anything over 120 characters. You can wrap this after any comma, just make sure to do so. Also don't forget to add extra indentation to the other half of the line, after you wrapped it.